src/shared/ui/survey/CheckBox/CheckBoxActivityItem.tsx

```diff
@@ -0,0 +1,76 @@
+import { FC, useEffect, useState } from 'react';
+
+import { shuffle } from '@shared/lib';
+import { Box, ScrollView } from '@shared/ui';
+
+import CheckBoxItem from './CheckBoxItem';
+import { Item } from './types';
+
+type CheckBoxProps = {
+  // @todo make sure backend will update config to new keys described below (type RefactoredConfig in ./types)
+  config: {
+    itemList: Item[];
+    minValue: number;
+    maxValue: number;
+    colorPalette: boolean;
+    randomizeOptions: boolean;
+  };
+  onChange: (arrayOfValues: number[]) => void;
+  initialValues?: number[];
+};
+
+const CheckBoxActivityItem: FC<CheckBoxProps> = ({
+  config,
+  onChange,
+  initialValues = [],
+}) => {
+  const [items, setItems] = useState<Item[]>([]);
+  const [answer, setAnswer] = useState<number[]>(initialValues);
```

Comment:
The component should be stateless. The state for it should be passed through props